### ISSUE TITLE
Add 2 new device types

### ISSUE THREAD

**Issue by afreof** [seveas/python-networkmanager#19]
NetworkManager 0.9.10+ supports new device types:
  * NM_DEVICE_TYPE_GENERIC = 14: The device is a generic interface type unrecognized by NetworkManager.
  * NM_DEVICE_TYPE_TEAM = 15: The device is a team master interface.

Here is a trivial patch adding two more constants. Thank you for applying it.
```
From 6bd6431f0bf5e30fb21c85d1d9fb920a7104b088 Mon Sep 17 00:00:00 2001
From: Adrian <adrian.freihofer@gmail.com>
Date: Wed, 4 Mar 2015 13:45:42 +0100
Subject: [PATCH] Add 2 new device types

---
 NetworkManager.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/NetworkManager.py b/NetworkManager.py
index b4b9aa1..f01bb76 100644
--- a/NetworkManager.py
+++ b/NetworkManager.py
@@ -422,6 +422,8 @@ NM_DEVICE_TYPE_BOND = 10
 NM_DEVICE_TYPE_VLAN = 11
 NM_DEVICE_TYPE_ADSL = 12
 NM_DEVICE_TYPE_BRIDGE = 13
+NM_DEVICE_TYPE_GENERIC = 14
+NM_DEVICE_TYPE_TEAM = 15
 NM_DEVICE_CAP_NONE = 0
 NM_DEVICE_CAP_NM_SUPPORTED = 1
 NM_DEVICE_CAP_CARRIER_DETECT = 2
-- 
1.9.1
```

**Comment by seveas** [comment]
I don't see these constant in any NetworkManager header file. Is this new since 0.9.8.8?

**Comment by afreof** [comment]
Thank you for the response.
Please search for "NM_DEVICE_TYPE_GENERIC" on https://developer.gnome.org/NetworkManager/0.9/spec.html[1] 
I'm currently using 0.9.10.1. NetworkManager uses type 14 for a tun device...

Regards Adrian

On Wednesday 04 March 2015 05.13:08 Dennis Kaarsemaker wrote:
> I don't see these constant in any NetworkManager header file. Is this new since 0.9.8.8?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/seveas/python-networkmanager/issues/19#issuecomment-77154949



--------------------
Adrian Freihofer
Gschwaderweg 29
8610 Uster
043 497 84 01

--------
[1] https://developer.gnome.org/NetworkManager/0.9/spec.html

**Comment by seveas** [comment]
Ok thanks!

The constants are usually updated by a script though. Could you do me a favor and do the following:

Delete all lines below 397 (the line that says the constants are generated)
run this in the git checkout: `python makeconstants.py >> NetworkManager.py` (you may need to install NetworkManager -dev/-devel packages for this to work)

And then post the diff? That will give me additional missing constants too :smile:

**Comment by afreof** [comment]
Of course. Here is the patch against NetworkManager 1.0

```
From 3bebcdbd928f1cbf34105a864b4035e70c32885f Mon Sep 17 00:00:00 2001
From: Adrian <adrian.freihofer@gmail.com>
Date: Wed, 4 Mar 2015 15:33:37 +0100
Subject: [PATCH] updated constants for NM 1.0

---
 NetworkManager.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/NetworkManager.py b/NetworkManager.py
index b4b9aa1..73789f8 100644
--- a/NetworkManager.py
+++ b/NetworkManager.py
@@ -422,9 +422,12 @@ NM_DEVICE_TYPE_BOND = 10
 NM_DEVICE_TYPE_VLAN = 11
 NM_DEVICE_TYPE_ADSL = 12
 NM_DEVICE_TYPE_BRIDGE = 13
+NM_DEVICE_TYPE_GENERIC = 14
+NM_DEVICE_TYPE_TEAM = 15
 NM_DEVICE_CAP_NONE = 0
 NM_DEVICE_CAP_NM_SUPPORTED = 1
 NM_DEVICE_CAP_CARRIER_DETECT = 2
+NM_DEVICE_CAP_IS_SOFTWARE = 4
 NM_WIFI_DEVICE_CAP_NONE = 0
 NM_WIFI_DEVICE_CAP_CIPHER_WEP40 = 1
 NM_WIFI_DEVICE_CAP_CIPHER_WEP104 = 2
@@ -529,6 +532,11 @@ NM_DEVICE_STATE_REASON_BR2684_FAILED = 51
 NM_DEVICE_STATE_REASON_MODEM_MANAGER_UNAVAILABLE = 52
 NM_DEVICE_STATE_REASON_SSID_NOT_FOUND = 53
 NM_DEVICE_STATE_REASON_SECONDARY_CONNECTION_FAILED = 54
+NM_DEVICE_STATE_REASON_DCB_FCOE_FAILED = 55
+NM_DEVICE_STATE_REASON_TEAMD_CONTROL_FAILED = 56
+NM_DEVICE_STATE_REASON_MODEM_FAILED = 57
+NM_DEVICE_STATE_REASON_MODEM_AVAILABLE = 58
+NM_DEVICE_STATE_REASON_SIM_PIN_INCORRECT = 59
 NM_DEVICE_STATE_REASON_LAST = 65535
 NM_ACTIVE_CONNECTION_STATE_UNKNOWN = 0
 NM_ACTIVE_CONNECTION_STATE_ACTIVATING = 1
-- 
1.9.1
```

**Comment by seveas** [comment]
Thanks, applied as 3c0ae35 and released as 0.9.13.

**Comment by afreof** [comment]
Thank you!
